packages/iTwinUI-react/src/core/Toast/Toaster.tsx

```diff
@@ -115,8 +107,40 @@ export default class Toaster {
     this.updateView();
   }
 
+  // Create container on demand.
+  // Cannot do it in constructor, because SSG/SSR apps would fail.
+  private async createContainer() {
+    const container = getContainer(TOASTS_CONTAINER_ID) ?? getDocument()?.body;
+    if (!container) {
+      return;
+    }
+
+    const toastWrapper = <ToastWrapper ref={this.toastsRef} />;
+
+    const _ReactDOM = ReactDOM as any; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+    // v18 mode
+    if (_ReactDOM.createRoot) {
+      // suppress warning about importing createRoot from react-dom/client
+      _ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.usingClientEntryPoint = true;
+
+      const root = _ReactDOM.createRoot(container);
+      root.render(toastWrapper);
+    } else {
+      // v17 and before
+      ReactDOM.render(toastWrapper, container);
+    }
+  }
+
   private updateView() {
-    this.toastsRef.current?.setToasts(this.toasts);
+    if (!this.isInitialized) {
+      this.createContainer().then(() => {
```

Comment:
Why `.then` if there is no awaits in `createContainer`?

Reply:
This is very interesting... Need to leave `createContainer` as async, otherwise toasts do not work on initial render (on first call).
My guess is it is connected with react life cycles and initial render of component. Without wrapping it as Promise (putting the execution as the last one in queue), it was setting the list of the toasts before it even got mounted and as a result, first call on `toaster` would render nothing, but then second one would show you 2 toasts instantly.